python/src/dataset.rs

```diff
@@ -972,8 +972,8 @@ impl Dataset {
         }
 
         for (key, value) in updates {
-            let column: &str = key.extract()?;
-            let expr: &str = value.extract()?;
+            let column: &str = &key.to_string();
+            let expr: &str = &value.to_string();
```

Comment:
```suggestion
            let column: &str = &key.downcast()?.to_string_lossy()?;
            let expr: &str = &value.downcast()?.to_string_lossy()?;
```

I worry that `to_string` does the right thing when it's a string but might do something strange if the python type isn't a string.